smplrout.cc

```diff
@@ -71,136 +72,75 @@
 #if FILTERS_ENABLED
 #define MYNAME "simplify"
 
-void SimplifyRouteFilter::free_xte(struct xte* xte_rec)
+inline bool operator<(const SimplifyRouteFilter::trackerror& lhs, const SimplifyRouteFilter::trackerror& rhs)
 {
-  delete xte_rec->intermed;
+  return ((lhs.dist > rhs.dist) || ((lhs.dist == rhs.dist) && (lhs.wptpos < rhs.wptpos)));
```

Comment:
We are defining operator<, not operator<=, so the tests don't include equality.
QMap returns items based on this operator from smallest to largest.  Since we want the smallest track error to be last I flipped the inequality in the dist comparison.  The polarity of the wptpos comparison is select to retain the same wpt as we did previously using stable_sort, flipping it results in a test failure.